### PR TITLE
Handle drafts and published - fix various issues

### PR DESCRIPTION
**Pull request by thebiggianthead**
Initially I created this branch to fix an issue where sort orders of published docs weren't updated if a draft doc existed. I ended up doing a few more fixes as I noticed things that weren't quite right, so this PR also covers a fair few things:

- updated `plugin-kit` and other deps
- when a draft and published document exists, only the draft document was patched with the new sort order, even though the plugin doesn't distinguish between draft and published documents. This meant it wasn't possible to reorder a published document in any situation a draft existed. Now, draft and published versions will be patched.
- Changed routing to use native Studio functionality. This means navigation between docs now happens without a page reload. Active documents are now given highlighted in the doc pane to give better context to users.
- Updated to use `sanity-plugin-utils` for the listener and for feedback components. This also solves as issue where a documents moving from draft to publish meant the title no longer displayed in the list pane.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -106,18 +97,19 @@ jobs:
...
-      - uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # tag=v3
+      - uses: actions/setup-node@v3
         with:
           cache: npm
           node-version: lts/*
       - run: npm ci
         # Branches that will release new versions are defined in .releaserc.json
-      - run: npx semantic-release
+        # @TODO remove --dry-run after verifying everything is good to go
+      - run: npx semantic-release --dry-run
```
Comment (snorrees) [inline .github/workflows/main.yml:112]: Dryrun needs to removed from here (todo)

**Review by snorrees — APPROVED** [review]
Awesome housekeeping! LGTM.

**Comment by thebiggianthead** [comment]
@SimeonGriggs Sorted. I've also added a counter on the drag when multiple items are being moved – I found it a little unclear given not all items are visibly moving so felt this might help...

**Comment by SimeonGriggs** [comment]
Nice touch, LGTM!

**Pull request merged**